### PR TITLE
Fixes #2087 - Show and hide navbar in dependency of scrolling

### LINKED ISSUE #2087: Top sticky header is breaking the anchors
Go to https://staging.webcompat.com/contributors/web-platform-research#surveys
capture d ecran 2018-02-16 a 08 01 27

We should be able to read the header of this section. 

Personally I don't think the header should be sticky. It should just go away when we scroll.

### PR DESCRIPTION
**Pull request by Regaddi**
I had to introduce a scrollHandler for this.

r? @miketaylr @zoepage

### CODE REVIEW

**Comment by zoepage** [comment]
LGTM!

File: webcompat/static/css/src/nav.css [hunk 1]
```diff
@@ -10,6 +10,12 @@
   box-sizing: border-box;
   background-color: var(--color-first);
   color: var(--color-second);
+  transition: transform 400ms ease, opacity 400ms ease;
```
Comment (miketaylr) [inline webcompat/static/css/src/nav.css]: Fancy transition. :)

File: webcompat/static/js/lib/homepage.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ function HomePage() {
   this.setUpEvents = function() {
     this.searchHandler();
     this.dropDownHandler();
+    this.navbarHandler();
```
Comment (miketaylr) [inline webcompat/static/js/lib/homepage.js]: I guess it feels a little weird to put this navbar controller code in homepage.js, but maybe it's more awkward we include homepage.js on every page. 😄 

We should probably file a followup bug to change that, and have something like navbar.js with fancy code-splitting. I imagine we'll move to a more modern modular design this year.

File: webcompat/static/js/lib/homepage.js [hunk 3]
```diff
@@ -63,6 +64,39 @@ function HomePage() {
...
+    var isScrolling = false;
+    $(window).on("scroll", function() {
+      isScrolling = true;
+      clearTimeout(scrollTimeout);
+      scrollTimeout = window.setTimeout(function() {
+        isScrolling = false;
+      }, 500);
+    });
+    window.setInterval(function() {
+      if (!isScrolling) return;
```
Comment (miketaylr) [inline webcompat/static/js/lib/homepage.js]: nit: can we add opening and closing braces and put the return on newline?

File: webcompat/static/js/lib/homepage.js [hunk 4]
```diff
@@ -63,6 +64,39 @@ function HomePage() {
...
 
+  this.navbarHandler = function() {
+    var $navbar = $(".navigation");
+    var navbarHeight = $navbar.outerHeight();
+    var lastScrollY = window.pageYOffset;
+    var scrollTimeout = null;
+    var isScrolling = false;
+    $(window).on("scroll", function() {
+      isScrolling = true;
+      clearTimeout(scrollTimeout);
```
Comment (miketaylr) [inline webcompat/static/js/lib/homepage.js]: I notice you're referring to the `setTimeout` and `setInterval` globals on the `window` object, but not `clearTimeout`. I don't have a hard preference either way (I normally just use global identifier w/o `window`), but it's probably good to be consistent.

Comment (Regaddi) [reply webcompat/static/js/lib/homepage.js]: I'd probably be explicit and always to refer to `window`.

File: webcompat/static/js/lib/homepage.js (hunk 3 above)
Comment (Regaddi) [reply webcompat/static/js/lib/homepage.js]: Sure.

File: webcompat/static/js/lib/homepage.js (hunk 2 above)
Comment (Regaddi) [reply webcompat/static/js/lib/homepage.js]: Indeed. I needed some time to find the "correct" place to add this :)

**Comment by miketaylr** [comment]
Filed https://github.com/webcompat/webcompat.com/issues/2268 as a follow up.

**Pull request merged**